lib/widgets/genres.dart

```diff
@@ -1,53 +1,89 @@
 import 'package:flutter/material.dart';
-import 'package:themoviedb/constants.dart';
-import 'package:themoviedb/custom_widgets.dart';
+
+import '../constants.dart';
+import '../custom_widgets.dart';
+import '../models/genre.dart';
+import '../repositories/genre_repository.dart';
 
 class Genres extends StatelessWidget {
-  const Genres({
+  Genres({
     super.key,
-    required this.genres,
+    required this.genresId,
+    required this.repository,
   });
 
-  final List<String> genres;
-  static const double genrePaddingHeight = 50.0;
-  static const double genrePaddingWidth = 100.0;
+  final List<int> genresId;
+  static const double genrePaddingHeight = 70.0;
+  static const double genrePaddingWidth = 120.0;
   static const double genresContainerHeight = 50.0;
+  final GenreRepository repository;
+  late Future<List<Genre>> allGenres;
+  late List<Genre> movieGenres;
 
   @override
   Widget build(BuildContext context) {
-    return Container(
-      height: genresContainerHeight,
-      alignment: Alignment.center,
-      child: ListView.builder(
-        itemCount: genres.length,
-        scrollDirection: Axis.horizontal,
-        itemBuilder: (BuildContext context, int index) {
-          return Padding(
-            padding: const EdgeInsets.all(Constants.generalPadding),
-            child: Container(
-              decoration: BoxDecoration(
-                color: Colors.deepPurple,
-                border: Border.all(
-                  color: Colors.blueGrey,
-                  width: Constants.boxDecorationWidth,
-                ),
-                borderRadius: const BorderRadius.all(
-                  Radius.circular(Constants.borderRadius),
-                ),
-              ),
-              padding: const EdgeInsets.all(Constants.generalPadding),
-              height: genrePaddingHeight,
-              width: genrePaddingWidth,
-              child: Center(
-                child: GeneralText(
-                  generalText: genres[index],
-                  fontSize: Constants.smallTextFont,
-                ),
-              ),
+    return FutureBuilder(
+      future: allGenres = repository.getData(),
```

Comment:
I'd follow the approach of the `Stateful` widget you did for the movies

Reply:
I understand the stateful approach, but I will use stateless. Next time I use FutureBuilder I will check whats the best option depending on the application

Reply:
changed to stateful. Done